packages/disputer/src/proxyTransactionWrapper.js

```diff
@@ -0,0 +1,183 @@
+  async submitDisputeTransaction(disputeArgs) {
+    // If the liquidator is not using a DSProxy, use the old method of liquidating
+    if (!this.useDsProxyToDispute) return await this._executeDisputeWithoutDsProxy(disputeArgs);
+    else return await this._executeDisputeWithDsProxy(disputeArgs);
+  }
+
+  async _executeDisputeWithoutDsProxy(disputeArgs) {
+    // Dispute strategy will control how much to liquidate
+    // Create the transaction.
+    const dispute = this.financialContract.methods.dispute(...disputeArgs);
+
+    // Send the transaction or report failure.
+    let receipt, returnValue;
+    try {
+      // Get successful transaction receipt and return value or error.
+      const transactionResult = await runTransaction({
+        transaction: dispute,
+        config: {
+          gasPrice: this.gasEstimator.getCurrentFastPrice(),
+          from: this.account,
+          nonce: await this.web3.eth.getTransactionCount(this.account)
+        }
+      });
+      receipt = transactionResult.receipt;
+      returnValue = transactionResult.returnValue.toString();
+    } catch (error) {
+      return error;
+    }
+
+    return {
+      type: "Standard EOA Dispute",
+      tx: receipt && receipt.transactionHash,
+      sponsor: receipt.events.LiquidationDisputed.returnValues.sponsor,
+      liquidator: receipt.events.LiquidationDisputed.returnValues.liquidator,
+      id: receipt.events.LiquidationDisputed.returnValues.liquidationId,
+      disputeBondPaid: receipt.events.LiquidationDisputed.returnValues.disputeBondAmount,
+      totalPaid: returnValue,
+      txnConfig: {
+        gasPrice: this.gasEstimator.getCurrentFastPrice(),
+        from: this.account
+      }
+    };
+  }
+
+  async _executeDisputeWithDsProxy(disputeArgs) {
+    const reserveCurrencyDisputer = new this.web3.eth.Contract(this.ReserveCurrencyDisputer.abi);
+
+    const callData = reserveCurrencyDisputer.methods
+      .swapDispute(
+        this.uniswapRouterAddress, // uniswapRouter
+        this.financialContract._address, // financialContract
+        this.reserveToken._address, // reserveCurrency
+        disputeArgs[0], // liquidationId
+        disputeArgs[1], // sponsor
+        this.maxReserverTokenSpent, // maxReserverTokenSpent
+        Number((await this.web3.eth.getBlock("latest")).timestamp) + this.tradeDeadline
+      )
+      .encodeABI();
+    const callCode = this.ReserveCurrencyDisputer.bytecode;
+
+    const dsProxyCallReturn = await this.dsProxyManager.callFunctionOnNewlyDeployedLibrary(callCode, callData);
+    const blockAfterDispute = await this.web3.eth.getBlockNumber();
+
+    // Wait exactly one block to fetch events. This ensures that the events have been indexed by your node.
+    await blockUntilBlockMined(this.web3, blockAfterDispute + 1);
+
+    const DisputeEvent = (
+      await this.financialContract.getPastEvents("LiquidationDisputed", {
+        fromBlock: blockAfterDispute - 1,
```

Comment:
Is there any chance that the event took place more than 1 block behind the `blockAfterDispute`?

Reply:
I dont see how that would be possible.